opentracing-apache-httpclient/src/main/java/io/opentracing/contrib/apache/ApacheClientSpanDecorator.java

```diff
@@ -0,0 +1,127 @@
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpRequestWrapper;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.protocol.HttpCoreContext;
+
+import io.opentracing.Span;
+import io.opentracing.tag.Tags;
+
+/**
+ * Decorate span at different stages of request processing. Do not finish span in decorator.
+ *
+ * @author Pavol Loffay
+ */
+public interface ApacheClientSpanDecorator {
+
+    /**
+     * Decorate span before request is fired.
+     *
+     * @param request request
+     * @param httpContext context
+     * @param span span to decorate
+     */
+    void onRequest(HttpRequestWrapper request, HttpContext httpContext, Span span);
+
+    /**
+     * Decorate span after response is received.
+     *
+     * @param response response
+     * @param httpContext context
+     * @param span span to decorate
+     */
+    void onResponse(HttpResponse response, HttpContext httpContext, Span span);
+
+    /**
+     * Decorate span after redirect.
+     *
+     * @param response response
+     * @param httpContext context
+     * @param span span to decorate
+     */
+    void onRedirect(HttpResponse response, HttpContext httpContext, Span span);
+
+    /**
+     *  Decorate span span on error e.g. {@link java.net.UnknownHostException}/
+     *
+     * @param request request
+     * @param httpContext context
+     * @param ex exception
+     * @param span span to decorate
+     */
+    void onError(HttpRequest request, HttpContext httpContext, Exception ex, Span span);
+
+    /**
+     * Decorator which adds standard set of tags and logs.
+     */
+    class StandardTags implements ApacheClientSpanDecorator {
+        private static final Logger log = Logger.getLogger(StandardTags.class.getName());
+
+        @Override
+        public void onRequest(HttpRequestWrapper request, HttpContext httpContext, Span span) {
+            Tags.COMPONENT.set(span, "apache-httpclient");
+            Tags.HTTP_METHOD.set(span, request.getRequestLine().getMethod());
+
+            URI uri = request.getURI();
+            Tags.HTTP_URL.set(span, request.getRequestLine().getUri());
+            Tags.PEER_PORT.set(span, uri.getPort() == -1 ? 80 : uri.getPort());
+            Tags.PEER_HOSTNAME.set(span, uri.getHost());
+        }
+
+        @Override
+        public void onResponse(HttpResponse response, HttpContext httpContext, Span span) {
+            Tags.HTTP_STATUS.set(span, response.getStatusLine().getStatusCode());
+        }
+
+        @Override
+        public void onRedirect(HttpResponse response, HttpContext httpContext, Span span) {
+            Map<String, Object> redirectLogs = new HashMap<>(4);
+            redirectLogs.put("event", "redirect");
+
+            Header locationHeader = response.getFirstHeader("Location");
+            if (locationHeader != null) {
+                redirectLogs.put("Location", locationHeader.getValue());
```

Comment:
Couldn't actually find it hence "seems" :) - but from looking at the values that have been defined on https://github.com/opentracing/specification/blob/master/semantic_conventions.md this does appear to be the case - but it should really be defined.